### PR TITLE
Basic RS232 Communication

### PR DESCRIPTION
**Pull request by DieKatzchen**
Uses a standard TTL to RS232 adapter. Feel free to rearrange pin assignments to other pins, uses SoftwareSerial to allow the use of the built in serial for debugging.

### CODE REVIEW

**Comment by 1technophile** [comment]
Hi,

So as to format automatically your code you have to add the "clang-Format" extension to VSCode, once done, you can format the code by doing a right-click into the code file window and clicking "Format document".

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 .pio
 .vscode
 *_env.ini
+main\User_config.h
+main/User_config.h
```
Comment (1technophile) [inline .gitignore:5]: I think we should remove this modification, indeed User_config.h needs to be stored on the configuration management.
If you have personal keys-password you can store them in a _env.ini file with platformio.

File: main/ZgatewayRS232.ino [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  if (cmpToMainTopic(topicOri, subjectMQTTtoRS232)) {
+    Log.trace(F("MQTTtoRS232" CR));
+    Log.trace(F("Prefix set: %s" CR), RS232Pre);
+    Log.trace(F("Postfix set: %s" CR), RS232Post);
+    RS232Serial.print(RS232Pre);
+    RS232Serial.print(datacallback);
+    RS232Serial.print(RS232Post);
+  }
+}
+#  endif
```
Comment (1technophile) [inline main/ZgatewayRS232.ino:84]: You can remove simpleReceiving, I will remove it on all the project for v0.9.6

**Review by 1technophile — COMMENTED** [review]
Thanks for the PR!

Would it be possible to update the documentation with this new gateway?

To do that you may add an RS232.md file [here](https://github.com/1technophile/OpenMQTTGateway/tree/development/docs/setitup) for the setup, and [here ](https://github.com/1technophile/OpenMQTTGateway/tree/development/docs/use)for the use
and link them from the [menu](https://github.com/1technophile/OpenMQTTGateway/blob/546b27773f990fa33fa457817a647f49accd098f/docs/.vuepress/config.js#L48)

**Comment by DieKatzchen** [comment]
Alright, I have some free time in *checks watch* two hours. I'll see about making those changes then.

File: .gitignore (hunk 1 above)
Comment (DieKatzchen) [reply .gitignore:5]: Whoops, meant to take that back out before the pull request. Thanks for the tip about the _env.ini, I'll try that. This is my first time using platformio.

**Comment by DieKatzchen** [comment]
I got stuck on the phone for almost two hours with Karen, who is true to her name, and have lost all energy and will to live. I shall do it tomorrow.

**Comment by 1technophile** [comment]
Lol  !!!

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,3 @@
 .pio
 .vscode
-*_env.ini
+*_env.ini
```
Comment (1technophile) [inline .gitignore:3]: Could you restore the original file, this way this file will not be identified in this PR

**Review by 1technophile — COMMENTED** [review]
some minors comments

File: .gitignore (hunk 3 above)
Comment (DieKatzchen) [reply .gitignore:3]: How would I do that? I feel like it would involve a git command.

Comment (1technophile) [reply .gitignore:3]: No, if you take the original file and replace the current one by the original, commit and push it. It should not appear anymore here.

Comment (DieKatzchen) [reply .gitignore]: I figured it out by `git restore` from master before I saw your simpler answer

**Comment by 1technophile** [comment]
Thanks!

**Comment by orrpan** [comment]
For the platformIO, which softwareserial library do you use? 
I guessed and used this: 

`softwareserial= plerup/EspSoftwareSerial@^6.9.0`

```
lib_deps =
  ${libraries.softwareserial}
```

**Pull request merged**